### ISSUE TITLE
Support for OpenAI's GPT-4o

### ISSUE THREAD

**Issue by shobees-dev** [OvidijusParsiunas/deep-chat#201]
We are in the process of integrating Deep Chat into our app and wish to utilize OpenAI's GPT-4o due to its new features and improvements. However, it appears that GPT-4o is not supported. This leaves us uncertain about whether we should proceed with the integration of Deep Chat.

Could the maintainers please provide insights on the current compatibility status with GPT-4o and any planned updates to include support for it? Your prompt response and continuous efforts to enhance Deep Chat are highly appreciated.

**Comment by OvidijusParsiunas** [comment]
Hi @shobees-dev.
Deep Chat does support the new [`GPT-4o`](https://platform.openai.com/docs/models/gpt-4o) model for both text and image communication. All you need to do is specify the model name in the [OpenAI Chat](https://deepchat.dev/docs/directConnection/OpenAI#Chat) property. For example:

```
directConnection = {
  openAI: {
    chat: { model: 'gpt-4o' },
    key: 'your-key'
  }
}
```

Let me know if you experience any issues. Thanks!

**Comment by OvidijusParsiunas** [comment]
To note, we will be making the `gpt-4o` model as the default model in our future release, so you will no longer need to specify it as so:

```
directConnection = {
  openAI: { key: 'your-key' }
}
```

This should be announced in the next month.

**Comment by shobees-dev** [comment]
Hi @OvidijusParsiunas,

Thank you for your quick reply. We are using OpenAI's Assistants API for text, image, and file inputs. However, when we use GPT-4o with the Assistants API, it doesn't seem to work. The responseInterceptor always displays the GPT-4 model.

```
directConnection={{
        openAI: {
          assistant: { model: "gpt-4o", function_handler: functionHandler },
          key: OPENAI_API_TEST_KEY,
        },
      }}
```
![image](https://github.com/OvidijusParsiunas/deep-chat/assets/168747262/28bc3f5f-0eaa-4ac6-b88b-9f743d090431)

**Comment by OvidijusParsiunas** [comment]
When it comes to OpenAI assistants, the model needs to be specified when an assistant is _created_.

You can do this in 2 ways:

1. Use the [`new_assistant`](https://deepchat.dev/docs/directConnection/OpenAI#NewAssistant) object and specify the model there. It is important to note that a new assistant will be created when a new conversation has started, hence a long term solution is recommended in option 2:
```
directConnection = {
  openAI: {
    assistant: { new_assistant: { model: 'gpt-4o' } },
    key: 'your-key'
  }
}
```
2. Use an already existing assistant that has a `gpt-4o` model assigned to it.
You can either create a new one in the [OpenAI Playground](https://platform.openai.com/playground/assistants) as follows:
<img width="1049" alt="Screenshot 2024-05-18 at 14 09 56" src="https://github.com/OvidijusParsiunas/deep-chat/assets/18709577/05691951-3f66-4f96-a8c4-7176a6c70829">

Or you can find the one you created in Option 1.
And then specify it using the [`assistant_id`](https://deepchat.dev/docs/directConnection/OpenAI#Assistant) property:

```
directConnection = {
  openAI: {
    assistant: { assistant_id: 'yout-assistant-id' },
    key: 'your-key'
  }
}
```

Let me know if this helps. Thanks!

**Comment by OvidijusParsiunas** [comment]
That is interesting, it appears that OpenAI have limited the assistants to be able to use the new model only in their version 2. I am currently working on migrating the Deep Chat [OpenAI Assistant](https://deepchat.dev/docs/directConnection/OpenAI#Assistant) code from v1 and v2. The real differences between the versions are actually just to do with files and not the text chat itself. Therefore if you want to just use text, you can actually use a temporary workaround where you can overwrite the request header containing the version via [`request headers`](https://deepchat.dev/docs/connect#Request) property, e.g.:

```
request = {
  headers: {
    'OpenAI-Beta': 'assistants=v2'
  }
}
```

Let me know if this works for you.

For the _file_ integration, please bare with me as I should have the code ready in `dev` soon. You can also track the issue [here](https://github.com/OvidijusParsiunas/deep-chat/issues/190).

Thanks!!

**Comment by OvidijusParsiunas** [comment]
Hi, the functionality mentioned in this issue is now available in Deep Chat version `2.0.0`.
